### PR TITLE
docs(core): docs fixes

### PR DESCRIPTION
**Pull request by vsavkin**

### CODE REVIEW

File: modules/@angular/core/src/metadata.ts [hunk 1]
```diff
@@ -72,19 +72,6 @@ export interface NgModuleDecorator extends TypeDecorator {}
...
- *
- * ### Example as ES5 annotation
- *
- * ```
- * var MyDirective = function() {
- *   ...
- * };
- *
- * MyDirective.annotations = [
- *   new ng.Directive({...})
```
Comment (vicb) [inline modules/@angular/core/src/metadata.ts]: why removing this ?

File: modules/@angular/router/src/config.ts [hunk 2]
```diff
@@ -253,11 +278,16 @@ export type ResolveData = {
 };
 
 /**
+ * @whatItDoes The type of `loadChildren`.
```
Comment (kara) [inline modules/@angular/router/src/config.ts]: This seems like a paste error? Shouldn't it be the type of the callback?

File: modules/@angular/router/src/directives/router_link.ts [hunk 3]
```diff
@@ -16,25 +16,38 @@ import {UrlTree} from '../url_tree';
...
- * When linking to this `User` route, you can write:
+ * When linking to this `user/:name` route, you can write:
+ *
+ * ```
+ * <a routerLink='/user/bob'>link to user component</a>
+ * ```
+ *
+ * @description
+ *
+ * The RouterLink directives lets you link to specific parts of your app.
```
Comment (kara) [inline modules/@angular/router/src/directives/router_link.ts]: lets -> let

File: modules/@angular/router/src/directives/router_link_active.ts [hunk 4]
```diff
@@ -16,6 +16,16 @@ import {RouterLink, RouterLinkWithHref} from './router_link';
 
 
 /**
+ * @whatItDoes Lets you add a CSS class to an element when the link's route.
```
Comment (kara) [inline modules/@angular/router/src/directives/router_link_active.ts]: I think you are missing `becomes active`

File: modules/@angular/router/src/router_module.ts [hunk 5]
```diff
@@ -23,15 +23,20 @@ import {flatten} from './utils/collection';
 
 
 /**
+ * @whatItDoes Contains a list of directives
  * @stable
  */
 const ROUTER_DIRECTIVES = [RouterOutlet, RouterLink, RouterLinkWithHref, RouterLinkActive];
 
 /**
+ * @whatItDoes Is used in DI to configure router.
```
Comment (kara) [inline modules/@angular/router/src/router_module.ts]: router -> the router

File: modules/@angular/router/src/url_tree.ts [hunk 6]
```diff
@@ -152,33 +245,57 @@ export function mapChildrenIntoArray<T>(
...
+ * @description
+ *
+ * Example URLs:
+ *
+ * ```
+ * /inbox/33(popup:compose)
+ * /inbox/33;open=true/messages/44
+ * ```
+ *
+ * DefaultUrlSerializer uses parenthesis to serialize secondary segments (e.g., popup:compose), the
```
Comment (kara) [inline modules/@angular/router/src/url_tree.ts]: parenthesis -> parentheses

File: modules/@angular/router/testing/router_testing_module.ts [hunk 7]
```diff
@@ -64,6 +89,11 @@ export function setupTestingRouter(
  * });
  * ```
  *
+ * @description
+ *
+ * The modules sets up the router to be used for testing.
+ * It provides spy implementations of Location, LocationStrategy, and NgModuleFactoryLoader.
```
Comment (kara) [inline modules/@angular/router/testing/router_testing_module.ts]: Do you want to link to these with @link?

**Comment by kara** [comment]
Found some typos, but other than that, LGTM

File: modules/@angular/router/src/config.ts [hunk 8]
```diff
@@ -232,11 +235,32 @@ import {Observable} from 'rxjs/Observable';
...
+ * loading them on demand. The router is designed to make lazy loading simple and easy. Instead of
+ * providing the children property, you can provide
+ * the loadChildren property, as follows:
+ *
+ * ```
+ * [{
+ *   path: 'team/:id',
+ *   component: Team,
+ *   loadChildren: 'team'
+ * }]
```
Comment (vicb) [inline modules/@angular/router/src/config.ts]: ;

File: modules/@angular/router/src/directives/router_link.ts [hunk 9]
```diff
@@ -65,7 +78,12 @@ import {UrlTree} from '../url_tree';
  * For instance, if the current url is `/user/(box//aux:team)`.
  *
  * Then the following link `<a [routerLink]="['/user/jim']">Jim</a>` will generate the link
- * `/user/(jim//aux:team)`. See {@link Router.createUrlTree} for more information.
+ * `/user/(jim//aux:team)`.
+ *
+ * @selector ':not(a)[routerLink]'
```
Comment (vicb) [inline modules/@angular/router/src/directives/router_link.ts]: hopefully this is extracted ?

**Pull request merged**